### ISSUE TITLE
SBOM generation

### ISSUE THREAD

**Issue by BobyMCbobs** [siderolabs/talos#7088]
### Description

SBOMs are an important way to prove what images and programs are made up of.

They can be generated with
- https://github.com/moby/buildkit/blob/master/docs/attestations/sbom.md
- https://github.com/anchore/syft

then signed and attested with `cosign` and stored in the container registry as tags
https://docs.sigstore.dev/cosign/attestation/

related: https://github.com/siderolabs/talos/issues/7087

keen to hear thoughts!

**Comment by smira** [comment]
Yep, I fully agree we should enable the SBOMs, as it should be an easy step. There was some debate internally on whether `cosign` is a good approach from security perspective, but it should anyways be better than nothing.

**Comment by BobyMCbobs** [comment]
In terms of generating the SBOMs, would something like https://github.com/moby/buildkit/blob/master/docs/attestations/sbom.md be useful or `go version -m <PROGRAM>`?

Then the sboms can be attested with 

```
cosign attest --predicate /path/to/sbom-spdx.json ghcr.io/siderolabs/talos:tag -y
```

**Comment by smira** [comment]
SBOM is part of the manifest, so `cosign` will sign it with the image itself.

We need to propagate SBOMs from `pkgs` which is something we should look into, but it won't be trivial.

**Comment by lucsoft** [comment]
Is this something for 2025?

**Comment by smira** [comment]
> Is this something for 2025?

quite possibly

**Comment by lucsoft** [comment]
nice work on the sbom's i see that these are made from syft, would it be possible to also provide cyclonedx files?

**Comment by smira** [comment]
> nice work on the sbom's i see that these are made from syft, would it be possible to also provide cyclonedx files?

No, we don't plan to provide other formats, but you can convert SPDX files to whatever format works better for you.

**Comment by lucsoft** [comment]
@smira yeah im kinda looking for a good workflow that allows that, the official converter from cyclonedx doesn't support SPDX 2.3 yet

**Comment by smira** [comment]
> [@smira](https://github.com/smira) yeah im kinda looking for a good workflow that allows that, the official converter from cyclonedx doesn't support SPDX 2.3 yet

I don't have an answer for that, it probably should go to the SPDX/Cyclone community. If we were to convert ourselves we'd hit the same issue. 

The scan process is [documented here](https://www.talos.dev/v1.11/advanced/sbom/#scanning-sboms).
